app/controllers/api/functions.php

```diff
@@ -382,13 +395,46 @@
     ->inject('response')
     ->inject('dbForInternal')
     ->inject('deletes')
-    ->action(function ($functionId, $response, $dbForInternal, $deletes) {
+    ->inject('project')
+    ->action(function ($functionId, $response, $dbForInternal, $deletes, $project) {
         /** @var Appwrite\Utopia\Response $response */
         /** @var Utopia\Database\Database $dbForInternal */
         /** @var Appwrite\Event\Event $deletes */
 
         $function = $dbForInternal->getDocument('functions', $functionId);
 
+        // Request executor to delete tag containers
+        $ch = \curl_init();
+        \curl_setopt($ch, CURLOPT_URL, "http://appwrite-executor:8080/v1/cleanup/function");
```

Comment:
For security reasons, we have an executor key. Not really sure why we need that, but since we have that, we shouldn't send it over HTTP, right? I might have misunderstood the concept of executor key.

Reply:
I see it in the headers, may be it was added later you added your comment.